test/test_emscripten.py

```diff
@@ -1,20 +1,44 @@
 from __future__ import annotations
 
-import platform
 import shutil
+import sys
 import textwrap
 
 import pytest
 
+from cibuildwheel.util import CIBW_CACHE_PATH
+
 from . import test_projects, utils
 
 basic_project = test_projects.new_c_project()
+basic_project.files["check_node.py"] = r"""
+import sys
+import shutil
+from pathlib import Path
+from pyodide.code import run_js
+
+
+def check_node():
+    node = shutil.which("node")
+    assert node is not None, "node is None"
+    node_path = Path(node).resolve(strict=True)
+    cibw_cache_path = Path(sys.argv[1]).resolve(strict=True)
+    assert cibw_cache_path in node_path.parents, f"{cibw_cache_path} not a parent of {node_path}"
+    node_js = run_js("globalThis.process.execPath")
+    assert node_js is not None, "node_js is None"
+    node_js_path = Path(node_js).resolve(strict=True)
+    assert node_js_path == node_path, f"{node_js_path} != {node_path}"
```

Comment:
Could you add comments to explain what is going on here? This is pretty confusing.

Reply:
done